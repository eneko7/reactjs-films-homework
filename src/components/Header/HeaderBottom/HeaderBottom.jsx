import React from 'react';
import MovieDescription from './MovieDescription/MovieDescription';
import MovieRate from './MovieRate/MovieRate';
import MovieInfoButtons from './MovieInfoButtons/MovieInfoButtons';
import style from './HeaderBottom.scss';

const HeaderBottom = () => (
  <div className={style.header_bottom}>
    <div className={`${style.active} ${style.header_bottom_left}`}>
      <MovieDescription filmName="THE JUNGLE BOOK" filmDescr="Adventure, Drama, Family, Fantasy, |, 1h 46m" />
      <MovieRate rate="4.8" />
    </div>
    <MovieInfoButtons annotation="There are growing dangers in the wizarding world og 1926 New York. Something mysterious is leaving a path of destruction in the streets, threatening to expose the wizarding" />
  </div>
);

export default HeaderBottom;
