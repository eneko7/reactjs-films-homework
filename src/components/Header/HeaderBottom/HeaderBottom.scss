.header_bottom {
  position: absolute;
  z-index: 3;
  bottom: 0;
  display: flex;
  margin-left: 7%;
  margin-bottom: 30px;
  width: 88%;
  justify-content: space-between;

  .header_bottom_left {
    position: relative;
    bottom: 0;
  }

  .header_bottom_left.active {
    position: static;
    padding-top: 204px;
  }
}

@media (max-width: 767px) {
  .header_bottom {
    display: block;
    width: 94%;
    margin-left: 6%;
    margin-bottom:50px;

    .header_bottom_left.active {
      padding-top: 0;
      margin-bottom: 20px;
    }
  }
}
