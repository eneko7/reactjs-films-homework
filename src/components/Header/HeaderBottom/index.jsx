export {dafault} from './HeaderBottom.jsx';