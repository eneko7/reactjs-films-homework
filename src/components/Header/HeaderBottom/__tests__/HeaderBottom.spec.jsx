import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import HeaderBottom from '../HeaderBottom';

const data = {
  genres: [
    {
      id: 18,
      name: 'Drama',
    },
    {
      id: 10751,
      name: 'Family',
    },
    {
      id: 14,
      name: 'Fantasy',
    },
  ],
  film: {
    id: 23355,
    original_title: 'name',
    title: 'name',
    genre_ids: [18, 10751, 14],
    vote_average: 5,
    overview: 'name',
  },
};
const badData = {
  genres: [
    {
      id: 18,
      name: 'Drama',
    },
    {
      id: 10751,
      name: 'Family',
    },
    {
      id: 14,
      name: 'Fantasy',
    },
  ],
  film: {},
};

describe('HeaderBottom', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<HeaderBottom {...data} />);
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
describe('HeaderBottom', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<HeaderBottom {...badData} />);
  it('renders correctly bad', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
