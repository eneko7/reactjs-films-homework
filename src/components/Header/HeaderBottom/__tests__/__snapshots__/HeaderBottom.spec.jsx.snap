// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBottom renders correctly 1`] = `
<div
  className="header_bottom"
>
  <div
    className="active header_bottom_left"
  >
    <MovieDescription
      filmGenres={
        Array [
          "Drama",
          "Family",
          "Fantasy",
        ]
      }
      filmName="name"
    />
    <MovieRate
      rate={5}
    />
  </div>
  <MovieInfoButtons
    annotation="name"
    filmId={23355}
  />
</div>
`;

exports[`HeaderBottom renders correctly bad 1`] = `null`;
